File: packages/sdk/src/config/environments.ts
import type { TheatricalEnvironment } from '../types/config';

const ENVIRONMENT_URLS: Record<TheatricalEnvironment, string> = {
  sandbox: 'https://api.sandbox.vista.co',
  staging: 'https://api.staging.vista.co',
  production: 'https://api.vista.co',
};

export function resolveBaseUrl(environment: TheatricalEnvironment, baseUrlOverride?: string): string {
  if (baseUrlOverride) return baseUrlOverride.replace(/\/$/, '');
  return ENVIRONMENT_URLS[environment];
}

export function isValidEnvironment(value: string): value is TheatricalEnvironment {
  return value === 'sandbox' || value === 'staging' || value === 'production';
}

File: packages/sdk-csharp/src/Theatrical.Sdk/Environments.cs
namespace Theatrical.Sdk;

/// <summary>
/// Vista platform environment configuration.
/// </summary>
public enum VistaEnvironment
{
    /// <summary>Sandbox environment for development and testing.</summary>
    Sandbox,

    /// <summary>Staging environment for pre-production validation.</summary>
    Staging,

    /// <summary>Production environment — live Vista platform.</summary>
    Production
}

/// <summary>
/// Resolves Vista API base URLs from environment identifiers.
/// </summary>
public static class EnvironmentResolver
{
    /// <summary>
    /// Get the base URL for a Vista environment.
    /// </summary>
    public static string GetBaseUrl(VistaEnvironment environment) => environment switch
    {
        VistaEnvironment.Sandbox => "https://api.sandbox.vista.co",
        VistaEnvironment.Staging => "https://api.staging.vista.co",
        VistaEnvironment.Production => "https://api.vista.co",
        _ => throw new ArgumentOutOfRangeException(nameof(environment))
    };
}
